docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,430 @@
+    hdbscan.labels_, index=vector_df.index, columns=["cluster_label"]
+)
+label_df["cluster_label"].value_counts()
+```
+
+![user distribution by cluster label](/user_distrib-by-cluster_label.png)
+
+## Visualizing the data
+
+Let's try to interpret our dataset by visualizing it using UMAP.
+
+First, we'll fit the UMAP and transform our dataset:
+
+```python
+umap_transform = umap.UMAP(random_state=0, transform_seed=0, n_jobs=1, metric="cosine")
+umap_transform = umap_transform.fit(vector_df)
+umap_vectors = umap_transform.transform(vector_df)
+umap_df = pd.DataFrame(
+    umap_vectors, columns=["dimension_1", "dimension_2"], index=vector_df.index
+)
+umap_df = umap_df.join(label_df)
+```
+
+Next,we join our dataframes and create a chart, letting us visualize the UMAP-transformed vectors, and coloring them with cluster labels.
+
+```python
+umap_df = umap_df.join(label_df)
+alt.Chart(umap_df).mark_circle(size=8).encode(
+    x="dimension_1", y="dimension_2", color="cluster_label:N"
+).properties(
+    width=600, height=500, title="UMAP Transformed vectors coloured by cluster labels"
+).configure_title(
+    fontSize=16,
+    anchor="middle",
+).configure_legend(
+    strokeColor="black",
+    fillColor="#EEEEEE",
+    padding=10,
+    cornerRadius=10,
+    labelFontSize=14,
+    titleFontSize=14,
+).configure_axis(
+    titleFontSize=14, labelFontSize=12
+)
+```
+
+![cluster visualization](/cluster_visualization)
+
+
+The dark blue clusters (label -1) are outliers - not large or dense enough to form a distinct group. 
+The large number of blobs [better word?] results from the fact that 1/3 of the vector norm mass is made of not very many ad_creatives.
+Note - 2D (UMAP) visualizations often make some clusters look quite dispersed [scattered].
+
+## Understanding the cluster groups
+
+To understand our user clusters better, we can generate some activity histograms.
+
+First, we join user data with cluster labels, create separate DataFrames for each cluster, generates activity histograms for each cluster, and then concatenates these histograms into a single visualization.
+
+```python
+# activity histograms by cluster
+user_df = user_df.set_index("id").join(label_df)
+
+by_cluster_data = {
+    label: user_df[user_df["cluster_label"] == label]
+    for label in np.unique(hdbscan.labels_)
+}
+
+activity_histograms = [
+    alt.Chart(user_df_part)
+    .mark_bar()
+    .encode(x=alt.X("activity", bin=True), y="count()")
+    .properties(title=f"Activity histogram for cluster {label}")
+    for label, user_df_part in by_cluster_data.items()
+]
+
+alt.hconcat(*activity_histograms)
+```
+
+![histograms of clusters](/cluster_histograms.png)
+
+From our histograms, we can observe that:
+
+- outliers (cluster -1)... [@mor - what does "outliers are the most active relatively - active users are rare" mean?]
+- cluster 2 and cluster 3 users are quite similar, positive[?], but low activity
+- cluster 0 has the highest proportion of medium activity users
```

Comment:
@morkapronczay 
"positive" = ?

Reply:
yeah that doesn't make much sense here. Low activity is enough